frontend/src/views/SignupView.tsx

```diff
@@ -1,21 +1,39 @@
 import { useNavigate } from "react-router-dom";
-// import UserForm from "../components/UserForm";
+import UserForm from "../components/UserForm";
 
 export default function SignupView() {
     const navigate = useNavigate();
 
-    // const onSubmit = async (
-    //     userid: string,
-    //     password: string,
-    // ): Promise<boolean> => {
-    //     // TODO: if signup is successful, navigate the user to log in
-    //     return false;
-    // };
+    //Signup function to handle the user registration process
+    const onSubmit = async (userId: string, password: string): Promise<boolean> => {
+        try {
+            //API call to create a new account
+            const response = await fetch("api/v1/auth", {
+                method: "POST",
+                headers: {"Content-Type": "application/json"},
+                body: JSON.stringify({userId, password}),
+            });
+
+            //Check if signup was successful
+            if(response.ok){
+                //if so navigate the user to login page
+                navigate("api/v1/auth/login") //NEEDS TO BE CHANGED TO CORRECT API CALL
+                return true;
+            } else {
+                //Handle error response (if user exists)
+                navigate("api/vs/auth/login")
+                return false; //failed new creation
+            }
+        } catch (error) {
+            console.error("An error occurred during signup", error);
```

Comment:
Same as the other error. I think the error will show up in the console anyway, so we can debug that